server.go

```diff
@@ -639,6 +639,7 @@ func (s *Server) ReadTCPSocket() {
 			case <-s.shutdown:
 				// occurs when cleanly shutting down the server e.g. in tests; ignore errors
 				log.WithError(err).Info("Ignoring Accept error while shutting down")
+				return
```

Comment:
Right now it falls through the switch, but that seems clearer. Good suggestion! Fixing ...